### PR TITLE
[COASTAL-969] System time offset detected

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/COASTAL-969

### CODE REVIEW

File: Loop/Managers/LoopAppManager.swift [hunk 1]
```diff
@@ -177,7 +177,8 @@ class LoopAppManager: NSObject {
                                                    closedLoopStatus: closedLoopStatus,
                                                    cacheStore: cacheStore,
                                                    localCacheDuration: localCacheDuration,
-                                                   overrideHistory: overrideHistory
+                                                   overrideHistory: overrideHistory,
+                                                   systemTimeOffsetDetector: trustedTimeChecker
```
Comment (ps2) [inline Loop/Managers/LoopAppManager.swift]: Is LDM's collaboration with TrustedTimeChecker done as a protocol (`SystemTimeOffsetDetector`) here for testing purposes?  I don't see any tests, so wondering what does the protocol give us over direct collaboration (LDM having a reference to TrustedTimeChecker directly)?

Not a blocker. I just tend value simplicity over reusability/substitutability at this layer.

**Review by ps2 — APPROVED** [review]
LGTM. Just a question about the `SystemTimeOffsetDetector` abstraction.

File: Loop/Managers/LoopAppManager.swift (hunk 1 above)
Comment (nhamming) [reply Loop/Managers/LoopAppManager.swift]: I could go either way. The abstraction was just to be explicit about responsibility and allow for future refactoring. I'll update as suggested.

**Comment by nhamming** [comment]
@ps2 The discussion at standup resulting in showing a best guess at the pump time. To do this, the trusted time checker needs to provide the system time offset that could be used to guess at the pump time.

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**